apache/metron
# METRON-1143 [FEATURE BRANCH] [NO-MERGE until PR#747] Management UI Support for Parser Extensions

## Contributor Comments
This PR introduces Management UI support for parser extensions.  It is based on the currently unmerged [PR 747](https://github.com/apache/metron/pull/747), and will be rebased when that hits the feature branch.

I believe that with this, we have mvp functionality with the parser extensions, as the creation of kafka topics, the availability of storm slots etc are existing issues beyond the extensions in general.


The UI support adds a new page for listing the installed extensions and the parsers they contain, and allows the installation and removal of those extensions.

<img width="1403" alt="screen shot 2017-09-15 at 07 10 42" src="https://user-images.githubusercontent.com/1111551/30480791-93c0d3b6-99e9-11e7-9606-10e486183a4e.png">
<img width="1146" alt="screen shot 2017-09-15 at 07 10 55" src="https://user-images.githubusercontent.com/1111551/30480795-9b3b35dc-99e9-11e7-8b68-7fa5cdb7165e.png">
<img width="1408" alt="screen shot 2017-09-15 at 07 11 30" src="https://user-images.githubusercontent.com/1111551/30480803-a55748bc-99e9-11e7-90d1-1aba66f205c7.png">
<img width="1409" alt="screen shot 2017-09-15 at 07 11 44" src="https://user-images.githubusercontent.com/1111551/30480805-a8771842-99e9-11e7-81e7-44307dbd6337.png">
<img width="324" alt="screen shot 2017-09-15 at 07 11 58" src="https://user-images.githubusercontent.com/1111551/30480808-aae287c4-99e9-11e7-9fa5-9793744f01a3.png">
<img width="1397" alt="screen shot 2017-09-15 at 07 12 31" src="https://user-images.githubusercontent.com/1111551/30480811-af8ddd82-99e9-11e7-9036-41efd7b0b9e1.png">

## Testing
- Build a parser extension per: https://github.com/apache/metron/blob/feature/METRON-1136-extensions-parsers/metron-maven-archetypes/metron-parser-extension-archetype/README.md

round trip:
- run in full dev
- go to the extension page
- click the add button
- find the archive.tar.gz of your extension from the assembly/target
- install
- select the extension in the list with the checkbox
- actions -> delete

After install, the parser(s) installed with the extension will be available in the sensors list
After install, you can create a new sensor config based on that parser

You should be able to start your parser by using the same rest ( swagger instructions )
such as documented in the readme above.

## TODO:
I don't have tests for all the components
I have not looked at the end to end stuff and may need help there
The install dialog needs css styling, and I need help with that.




### For code changes:
  ```
  mvn -q clean integration-test install && build_utils/verify_licenses.sh 
  ```

- [na] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)? 

### For documentation related changes:

  ```
  cd site-book
  mvn site
  ```
